src/views/sign-up/SignUp.jsx

```diff
@@ -0,0 +1,493 @@
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				setLastName,
+			);
+
+		e.target.name === 'email' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				ValidateEmail,
+				setFormDataValid,
+				setEmail,
+			);
+
+		e.target.name === 'address' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				setAddress,
+			);
+
+		e.target.name === 'userName' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				setuserName,
+			);
+
+		e.target.name === 'telephone' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validateEmptyField,
+				setFormDataValid,
+				settelephone,
+			);
+		e.target.name === 'password' &&
+			changeHandler(
+				e,
+				{
+					firstName,
+					lastName,
+					userName,
+					address,
+					telephone,
+					email,
+					password,
+				},
+				validatePassword,
+				setFormDataValid,
+				setPassword,
+			);
+	}
+
+	function handleOnFocus(e) {
+		e.target.name === 'firstName' && focusHandler(setFirstName, firstName);
+		e.target.name === 'lastName' && focusHandler(setLastName, lastName);
+		e.target.name === 'address' && focusHandler(setAddress, address);
+		e.target.name === 'email' && focusHandler(setEmail, email);
+		e.target.name === 'userName' && focusHandler(setuserName, userName);
+		e.target.name === 'telephone' && focusHandler(settelephone, telephone);
+		e.target.name === 'password' && focusHandler(setPassword, password);
+	}
+
+	async function handleSubmit(e) {
+		try {
+			e.preventDefault();
+			const userData = {
+				firstName,
+				lastName,
+				userName,
+				address,
+				telephone,
+				email,
+				password,
+			};
+
+			const reqBody = Object.entries(userData).reduce((acc, curr) => {
+				return {
+					...acc,
+					[curr[0]]: curr[1].value,
+				};
+			}, {});
+
+			setErrorMsg('');
+			setCreatinAcc(true);
+
+			const res = await axios.post(
+				`${
+					process.env.LIVE_API_URL ||
+					'https://e-comm-team-amigos-bn-project.onrender.com'
+				}/user/create`,
+				reqBody,
+			);
+
+			if (res.status === 201) {
+				setCreatinAcc(false);
+				setAccCreated(true);
+			}
+		} catch (error) {
+			if (typeof error.response.data === 'string') {
+				setErrorMsg(error.response.data);
+				setCreatinAcc(false);
+			} else {
+				setCreatinAcc(false);
+				setErrorMsg(error.response.data.error);
+			}
+		}
+	}
```

Comment:
Good job for integration, but I'm suggesting to take advantage of using redux to handle integration functionality and will be easy for you to manage component states.